src/contrib/inference/abstractmcmc.jl

```diff
@@ -0,0 +1,87 @@
+struct TuringState{S,F}
+    state::S
+    logdensity::F
+end
+
+struct TuringTransition{T,NT<:NamedTuple,F<:AbstractFloat}
+    θ::T
+    lp::F
+    stat::NT
+end
+
+function TuringTransition(vi::AbstractVarInfo, t)
+    theta = tonamedtuple(vi)
+    lp = getlogp(vi)
+    return TuringTransition(theta, lp, getstats(t))
+end
+
+metadata(t::TuringTransition) = merge((lp = t.lp,), t.stat)
+DynamicPPL.getlogp(t::TuringTransition) = t.lp
+
+state_to_turing(f::DynamicPPL.LogDensityFunction, state) = TuringState(state, f)
+function transition_to_turing(f::DynamicPPL.LogDensityFunction, transition)
+    θ = getparams(transition)
+    varinfo = DynamicPPL.unflatten(f.varinfo, θ)
+    # TODO: `deepcopy` is overkill; make more efficient.
+    varinfo = DynamicPPL.invlink!!(deepcopy(varinfo), f.model)
+    return TuringTransition(varinfo, transition)
+end
+
+# NOTE: Only thing that depends on the underlying sampler.
+# Something similar should be part of AbstractMCMC at some point:
+# https://github.com/TuringLang/AbstractMCMC.jl/pull/86
+getparams(transition::AdvancedHMC.Transition) = transition.z.θ
+getstats(transition::AdvancedHMC.Transition) = transition.stat
+
+getparams(transition::AdvancedMH.Transition) = transition.params
+getstats(transition) = NamedTuple()
```

Comment:
@torfjelde Isn't this obsolete now that #2026 was merged?

Reply:
Ah gotcha. @JaimeRZP you can always just do `]add Turing#torfjelde/allow-abstractsampler-draft` if you want to try out recent developments. And if you want to develop features based on this branch, just create a branch based on this PR and then continue from there, as you did with #2028 :)